### ISSUE TITLE
no-missing-path doesn't support URL encoding

### ISSUE THREAD

**Issue by AJDurant** [ota-meshi/eslint-plugin-markdown-links#121]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

10.6.0

### What version of `eslint-plugin-markdown-links` are you using?

0.9.0

### What did you do?

```md
[Link to path that exists](../Path%20With%20Space/File%20With%20Space.md)
```

Description:
When using URL encoded paths for paths with spaces, the `no-missing-path` rules does not detect it as existing.

### What did you expect to happen?

Valid paths (with URL encoded spaces) would be detected as valid.

### What actually happened?

```txt
error  The file '../Path%20With%20Space/File%20With%20Space.md' does not exist. Please check the path or update it to a valid file
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNqFU12P2jAQ/CvbXCsFiTgtj1RXtbrSp0o9qY8Y6dx4CQbHjmyHckL8965NwpfuehHKhz07szNr9pl3VYk70bQaWSOzaXYHj1o81852RnLDzd0dPKyw2oC2ldCwVBqhFWHlQRgJSyfqBk3wEVrAfLZTPihTw4+IeyTcImeDwIfJxwmJjN5GQsJBJP2NlSUhrcwGlAdjAwjYCq0kNMJtpP1r0iY32fjSTaJIfmbHhUnvhgiDsgY+3XzTPjGgJ7LASHOpatasPVGoprUuwB4kLpXBh7QHBzJvG+B9SXks4dlnbsoS+ppTiz32aw8e1hP8BvuT3Phr8qLVXa1MMUCK6NinYpL1AVyncWaCU+jhHn79WZMpRnOJC/kVMYtQP2I0yIAu5wYgn48TwWIE91/gXXxlDQbBJLYOKxFQjrkZkRjuUqeUg+h0ep7yyOeRipwLSZOxDqFGg05ViRo8Bg8rdDgG31UrEH7a44fu+sypQZp4Q4dKJlm4zuVV1D7eIIn56fCVBL5RQ0LrN8M81g6FjLE+xjiJPtsU1/G6SJw1os3zuRENuTsHOX+6Vijf7yPk8DSOrThnHc8Wo9R8vM5v1LPdEkBJLGOaqa+YYPy/vJRivHh2o2ZsIVHIonOazE3Pmr3OIT3jfRFHS2e/FdVG1MjW3ho69ilDnoXnFo/1jZXUSU/AM4nb79jGCZiKUoiYPvaXznkk0HSUfDh1MAzkv5uvTeuyJBnhhn7Z4R9bvaQK

### Additional comments

Note that:

```md
[Link to path that exists](../Path With Space/File With Space.md)
```

Is not detected as an error, but it is not a valid markdown link.

Looking at the [commonmark spec](https://spec.commonmark.org/0.31.2/#links) these can be written as:

```md
[Link to path that exists](<../Path With Space/File With Space.md>)
```

This does work, so perhaps this is all working as intended.

**Comment by ota-meshi** [comment]
Thank you for the report!
I have fixed it.
